modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.layout.reports.web.internal.display.context;
+
+import com.liferay.layout.seo.kernel.LayoutSEOLink;
+import com.liferay.layout.seo.kernel.LayoutSEOLinkManager;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.portlet.RenderRequest;
+
+/**
+ * @author Cristina González
+ */
+public class LayoutReportsDisplayContext {
+
+	public LayoutReportsDisplayContext(
+		GroupLocalService groupLocalService,
+		LayoutLocalService layoutLocalService,
+		LayoutSEOLinkManager layoutSEOLinkManager, Language language,
+		Portal portal, RenderRequest renderRequest) {
+
+		_groupLocalService = groupLocalService;
+		_layoutLocalService = layoutLocalService;
+		_layoutSEOLinkManager = layoutSEOLinkManager;
+		_language = language;
+		_portal = portal;
+		_renderRequest = renderRequest;
+
+		_themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public Map<String, Object> getData() {
+		if (_data != null) {
+			return _data;
+		}
+
+		Layout layout = _layoutLocalService.fetchLayout(
+			_themeDisplay.getPlid());
+
+		_data = Collections.singletonMap(
+			"context",
+			HashMapBuilder.<String, Object>put(
+				"assetsPath",
+				_portal.getPathContext(_renderRequest) + "/assets/"
+			).put(
+				"canonicalURLs", _getCanonicalURLs(layout)
```

Comment:
I don't know if some of this attributes must be on `props` instead of `context`, I'm not sure. What do you think?

Reply:
I don't have idea how do you use props, context ... I will use the format it is better for you.

Reply:
The usage of `props` and/or `context` it's the same, I'm going to investigate a little bit the conditions to add an attribute in each of them, just curious...

Reply:
I think we are the only ones using `props` and `context` separately. Most of the `<react:component />` uses the `HashMapBuilder` directly in the JSP. Other way to do it is to call getData and in the method something like this: https://github.com/liferay/liferay-portal/blob/6265d795b50b5e686e73d7e967d8d3283be2e792/modules/apps/asset/asset-categories-selector-web/src/main/java/com/liferay/asset/categories/selector/web/internal/display/context/AssetCategoriesSelectorDisplayContext.java#L151-L165 (removing `context` at the first level). What do you think? 🤔

Reply:
We can do it in another PR, and applied the same format to AnalyticsReports, so we don't have incosistent APIs